### PR TITLE
Added interactiveness

### PR DESCRIPTION
**Pull request by ToberoCat**
# Interactiveness
This allows users to input any item into the anvil gui, which allows developers to use this api to get inputs from users not only as text.

## The changes include:
- The AnvilGui now uses a Function5 as the complete callback. The first argument is the player, second the text, third is the left item and fourth is the right one. No item can be null, will always be replaced with an air item.
- The AnvilBuilder now has an method overflow. This allows the new usage and the old one, therefore not breaking old guis created with it.
- Interactivw is a state that allows the developer to decide if the users is able to input items / take them out of it. It defaults to false, so it doesn't destroy any previous made guis

## Long story short:
- It got tested in 1.8 and 1.19.
- It doesn't destroy any old anvil guis that got created using the AnvilGui.Builder class.
- Updated the readme with the new methods
- Items will get dropped when the gui gets closed by the user. Items will get removed when the AnvilGui.Response.close response got triggered. This only happens when the interactive state is true

### CODE REVIEW

File: api/src/main/java/net/wesjd/anvilgui/AnvilGUI.java [hunk 1]
```diff
@@ -58,14 +65,20 @@ public class AnvilGUI {
...
     /**
      * An {@link Consumer} that is called when the anvil GUI is closed
      */
     private final Consumer<Player> closeListener;
     /**
-     * An {@link BiFunction} that is called when the {@link Slot#OUTPUT} slot has been clicked
+     * An {@link Function5} that is called when the {@link Slot#OUTPUT} slot has been clicked
      */
-    private final BiFunction<Player, String, Response> completeFunction;
+    private final Function5<Player, String, ItemStack, ItemStack, Response> completeFunction;
```
Comment (WesJD) [inline api/src/main/java/net/wesjd/anvilgui/AnvilGUI.java:81]: For the sake of sanity for the programmer, I think we should go ahead and make the `completeFunction` a simple `Function<Completion, Response>` where a `Completion` would contain all of the data like the player who clicked, etc.

This would be very scalable in the future if we need to make any other changes while avoiding another breaking change.

**Review by WesJD — CHANGES_REQUESTED** [review]
Overall a super clean impl. Either you or I can do the change that I requested. I will be home for thanksgiving this weekend and will be able to merge this in!

File: api/src/main/java/net/wesjd/anvilgui/AnvilGUI.java (hunk 1 above)
Comment (ToberoCat) [reply api/src/main/java/net/wesjd/anvilgui/AnvilGUI.java]: Okay, upload the suggested changes

**Comment by WesJD** [comment]
So, after reading it over more, here are a few issues I've found:
1. When the `preventClose` option is used, when the player closes the inventory after they have inputted items into the left or right slot, they get put back in their inventory instead of being placed back into the slots on reopen.
2. Shouldn't the developer be the one deciding whether the items should be given back to the player when the inventory is closed? If so, we should not be giving back items at all. (kind of overrides the first point)

Also, just overall, do you have an example reason for introducing this feature (which will likely require further maintenance)? Couldn't a developer simply use a chest inventory in order to obtain item input? Just trying to understand the use case.

**Comment by ToberoCat** [comment]
1. / 2. You can remove drop on close thing and it would me solved
3. They get the output anyways, so we don't have to handle that. The devs can just give the player the item like they would do with the left / right item

I was writing a plugin for a client, who didn't like me to use chest guis for some reason. He wanted me to use anvil guis for item inputting. That's why I modified your code, so it allows me to use the gui, without having to write my own nms system.

After I finished the plugin, I just thought I should make it a pull request, maybe some people have such clients too.

<details>
<summary>Example with code. Please note, this code is only a small portion of the code I used for the anvil gui input system. It isn't supposed to compile, only to give you the idea how the feature was used. Some of the text displayed to the user is german, because the plugin should be running on a german city build server</summary>

```java
public class RecipeWizard {

    public void wizard() {
        askRecipeId() // I didn't past every single method that asks for another item, only pasted one
                .then(this::askRecipeName)
                .then(this::askCategory)
                .then(this::askOre)
                .then(this::askNormalItem)
                .then(this::askFuel)
                .then(this::askResult)
                .then(this::createRecipe);
    }

    // ....

    private @NotNull TaskPromise<String> askOre() {
        return new TaskPromise<>(resolve -> new AnvilGUI.Builder()
                .onClose(u -> {
                    if (ore != null) return;
                    send("Der Item Erstellungsprozess wurde beendet");
                })
                .onComplete(completion -> {
                    ore = completion.getLeftItem();
                    resolve.accept(completion.getText());
                    return AnvilGUI.Response.close();
                })
                .canBeInteractedWith(true)
                .text("Erz")
                .itemLeft(createItem(Material.PAPER, ""))
                .title("Erz reinlegen")
                .plugin(CustomItems.getPlugin())
                .open(player));
    }

}
```

</details>

**Comment by WesJD** [comment]
Thanks for your help on this. I modified your original idea slightly to be more configurable by the developer.
